src/libcore/intrinsics.rs

```diff
@@ -625,8 +852,26 @@ extern "rust-intrinsic" {
     /// appropriate for operations on the same thread that may be preempted,
     /// such as when interacting with signal handlers.
     pub fn atomic_singlethreadfence();
+    /// A compiler-only memory barrier.
+    ///
+    /// Memory accesses will never be reordered across this barrier by the
+    /// compiler, but no instructions will be emitted for it. This is
+    /// appropriate for operations on the same thread that may be preempted,
+    /// such as when interacting with signal handlers.
```

Comment:
I haven't found any, but it is very likely that I missed some stabilized versions.
Because there are many intrinsics, what I did was to search for any reference to intrinsics in `libcore` and `libstd`. Now that the biggest part of the work is done, I can try to improve few documentations one by one

Reply:
As they are explicitely documented as "compiler-only", I expect there are none. But I could have missed some, will do some deeper research.

Reply:
Actually, there is `std::sync::atomic::compiler_fence`